design/general_refactorings.md

```diff
@@ -0,0 +1,49 @@
+Before refactor concrete thing here is a list of things that need to be done all over the code:
```

Comment:
I would start with the purpose of this document that you are creating

Please add titles, subtitles, etc. See our template [here](https://github.com/cyberark/secrets-provider-for-k8s/blob/master/design/templates/solution_design.md)

For example:
## CLI Refactor

The purpose of this document is to outline the parts of the Python CLI/SDK code that we would like to refactor as well as provide a design for each point. Through this refactor, we hope that the code will be more readable, and easier to contribute to and maintain, and follow best SOLID practices.

Reply:
Done